### PR TITLE
Give error for non-time-series editDatabaseType

### PR DESCRIPTION
**Pull request by ktarbet**
Avoids Null Pointer Exception when running alarmedit.bat  (AlarmEditor.java) with a xml database.  This will help other programs/batch-files give better error messages.

### CODE REVIEW

File: src/main/java/decodes/tsdb/TsdbAppTemplate.java [hunk 1]
```diff
@@ -105,511 +33,518 @@
...
+            return;
+
+//        String className = DecodesSettings.instance().dbClassName;
+        String className = DecodesSettings.instance().getTsdbClassName();
+
+        if (className == null)
+        {
+            String message = "Error: can't create time series database with editDatabaseType= '"
+                    + DecodesSettings.instance().editDatabaseType + "'";
+            Logger.instance().fatal(message);
```
Comment (MikeNeilson) [inline src/main/java/decodes/tsdb/TsdbAppTemplate.java:277]: We should change this to slf4j and using the string formatting features.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]

File: src/main/java/decodes/tsdb/TsdbAppTemplate.java (hunk 1 above)
Comment (ktarbet) [reply src/main/java/decodes/tsdb/TsdbAppTemplate.java:277]: If we construct a message (with concatenation) for purpose of a exception constructor message **and** logging, like the example above  -- Is there still  benefit to string formatting?

Other cases in this file are more clear cut and the slf4j {} formatting is clearly helpful.

Comment (MikeNeilson) [reply src/main/java/decodes/tsdb/TsdbAppTemplate.java:277]: If I'm doing both I'll usually use String.format. But I've run into this problem myself, should it even do both? Or just throw the exception?

File: src/main/java/decodes/tsdb/TsdbAppTemplate.java [hunk 2]
```diff
@@ -384,24 +369,14 @@ public void tryConnect()
...
+     * @param afn auth filename
+     * @param ex exception
      */
     protected void authFileEx(String afn, Exception ex)
     {
-        String msg = "Cannot read DB auth from file '" + afn + "': " + ex;
-        System.err.println(msg);
-        Logger.instance().failure(msg);
-try { throw new Exception(""); } catch (Exception ex2) { ex2.printStackTrace(); }
+        log.error("Cannot read DB auth from file '{}'", afn ,ex);
```
Comment (MikeNeilson) [inline src/main/java/decodes/tsdb/TsdbAppTemplate.java:376]: I believe this form doesn't work and the .atError().setCause(ex).log(...) needs to be used.
But easy to test with a bad auth file.

**Review by MikeNeilson — COMMENTED** [review]
Looking good. Thanks for adjusting the whole file.

File: src/main/java/decodes/tsdb/TsdbAppTemplate.java (hunk 1 above)
Comment (ktarbet) [reply src/main/java/decodes/tsdb/TsdbAppTemplate.java]: Thanks, I'm trying String.format

**Review by MikeNeilson — APPROVED** [review]

File: src/main/java/decodes/tsdb/TsdbAppTemplate.java [hunk 3]
```diff
@@ -340,42 +261,40 @@ public synchronized void createDatabase()
 		throws ClassNotFoundException,
 		InstantiationException, IllegalAccessException
 	{
-		if (theDb != null)
+		if (theDb != null) {
```
Comment (MikeNeilson) [inline src/main/java/decodes/tsdb/TsdbAppTemplate.java:264]: { on next line.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
One nitpicky bracket formatting issue.

**Review by MikeNeilson — APPROVED** [review]

**Pull request merged**